### PR TITLE
FieldTmp Operations: Doxygen Strings

### PR DESCRIPTION
**Pull request by ax3l**
- [x] follow-up to #1788 (needs rebase)

Adds doxygen strings to the classes that users need to define `FieldTmp` operations in `fileOutput.param` or algorithms such as Thomas Fermi. This can help to understand how to use those and can be more detailed then the small examples we give in `fileOutput.param`.

### CODE REVIEW

File: src/picongpu/include/particles/particleToGrid/ComputeGridValuePerFrame.def [hunk 1]
```diff
@@ -77,10 +84,17 @@ public:
...
+ * Each value is mapped per cell according to the species' spatial shape.
+ *
+ * @tparam T_Species a @see picongpu::Particles class with a species definition,
+ *                   see @see speciesDefinition.param
+ *
+ * @typedef CreateDensityOperation< T_Species >::type a field that can be used
+ *          in @see fileOutput.param @see picongpu::FileOutputFields
  */
-
-/* Density */
```
Comment (PrometheusPi) [inline src/picongpu/include/particles/particleToGrid/ComputeGridValuePerFrame.def]: QUESTION:
Why did you remove this comment?

File: src/picongpu/include/particles/particleToGrid/ComputeGridValuePerFrame.def [hunk 2]
```diff
@@ -93,7 +107,21 @@ struct CreateDensityOperation
...
 };
 
-/* ChargeDensity */
+/** Charge Density Operation for Particle to Grid Projections
+ *
+ * Derives a scalar charge density field from a particle species at runtime.
+ * Each value is mapped per cell according to the species' spatial shape.
+ *
+ * @note for species that do not change their charge state, this is identical
+ *       to the density times a constant for the charge,
```
Comment (PrometheusPi) [inline src/picongpu/include/particles/particleToGrid/ComputeGridValuePerFrame.def]: OPTIONAL:
better: `a constant for the charge` -> `the (constant) particle's charge`

File: src/picongpu/include/particles/particleToGrid/ComputeGridValuePerFrame.def [hunk 3]
```diff
@@ -119,7 +162,25 @@ struct CreateMidCurrentDensityComponentOperation
     typedef FieldTmpOperation< ParticleMidCurrentDensityComponent, T_Species > type;
 };
 
-/* ParticleCounter */
+/** Real-Particle Counter Operation for Particle to Grid Projections
+ *
+ * Derives a scalar field with real particle numbers per cell from a particle
+ * species at runtime.
+ * Each macro particles weighting is assigned straight to the cell it belongs
+ * to, which is in most cases as floor operation in space (and not necessarily
```
Comment (PrometheusPi) [inline src/picongpu/include/particles/particleToGrid/ComputeGridValuePerFrame.def]: REQUIRED:
typo: `as`-> `a`

File: src/picongpu/include/particles/particleToGrid/ComputeGridValuePerFrame.def [hunk 4]
```diff
@@ -143,7 +219,22 @@ struct CreateMacroCounterOperation
     typedef FieldTmpOperation< MacroParticleCounter, T_Species > type;
 };
 
-/* EnergyDensity */
+/** Energy Density Operation for Particle to Grid Projections
+ *
+ * Derives a scalar field for average kinetic particle energy per cell times the
+ * particle density from a particle species at runtime.
```
Comment (PrometheusPi) [inline src/picongpu/include/particles/particleToGrid/ComputeGridValuePerFrame.def]: QUESTION: 
Would `kinetic particle energy density` be not more straight forward?

File: src/picongpu/include/particles/particleToGrid/ComputeGridValuePerFrame.def (hunk 1 above)
Comment (ax3l) [inline src/picongpu/include/particles/particleToGrid/ComputeGridValuePerFrame.def]: it's not removed, there is a whole new block now with a real doxygen comment for this class.

File: src/picongpu/include/particles/particleToGrid/ComputeGridValuePerFrame.def (hunk 2 above)
Comment (ax3l) [inline src/picongpu/include/particles/particleToGrid/ComputeGridValuePerFrame.def]: thanks, but better plural: `the (constant) particles' charge`

File: src/picongpu/include/particles/particleToGrid/ComputeGridValuePerFrame.def (hunk 3 above)
Comment (ax3l) [inline src/picongpu/include/particles/particleToGrid/ComputeGridValuePerFrame.def]: thx, fixed

File: src/picongpu/include/particles/particleToGrid/ComputeGridValuePerFrame.def (hunk 4 above)
Comment (ax3l) [inline src/picongpu/include/particles/particleToGrid/ComputeGridValuePerFrame.def]: nope, the term energy density is quite well defined to be kinetic. If someones does not get that in the title the brief directly explains it in detail.

File: src/picongpu/include/particles/particleToGrid/ComputeGridValuePerFrame.def [hunk 5]
```diff
@@ -119,7 +162,25 @@ struct CreateMidCurrentDensityComponentOperation
     typedef FieldTmpOperation< ParticleMidCurrentDensityComponent, T_Species > type;
 };
 
-/* ParticleCounter */
+/** Real-Particle Counter Operation for Particle to Grid Projections
+ *
+ * Derives a scalar field with real particle numbers per cell from a particle
+ * species at runtime.
+ * Each macro particles weighting is assigned straight to the cell it belongs
```
Comment (ax3l) [inline src/picongpu/include/particles/particleToGrid/ComputeGridValuePerFrame.def]: updated

File: src/picongpu/include/particles/particleToGrid/ComputeGridValuePerFrame.def [hunk 6]
```diff
@@ -131,7 +192,49 @@ struct CreateCounterOperation
     typedef FieldTmpOperation< ParticleCounter, T_Species > type;
 };
 
-/* EnergyDensity */
+/** Macro-Particle Counter Operation for Particle to Grid Projections
+ *
+ * Derives a scalar field with number of macro particles per cell from a particle
+ * species at runtime.
+ * Each macro particles is counted straight to the cell it belongs
```
Comment (ax3l) [inline src/picongpu/include/particles/particleToGrid/ComputeGridValuePerFrame.def]: thx, updated
